src/libPMacc/include/archAlignment.hpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2016 Rene Widera
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include <boost/align/alignment_of.hpp>
+
+namespace PMacc
+{
+    /** object to test for a useful alignment
+     *
+     * The compiler auto align the member array to a useful architecture
+     * depending value.
+     */
+    struct UsefulAlignTestObject
+    {
+        char x[512] __attribute__ ((aligned));
```

Comment:
probably using the `BOOST_ALIGNMENT` macro is a portable way to do this.

btw, also a [Boost.Align](http://www.boost.org/doc/libs/1_61_0/doc/html/align.html) exists (1.56+?).